File: src/main/java/no/novari/discovery/service/model/dtos/InstanceValueMetadataDto.java
package no.novari.discovery.service.model.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;
import no.novari.discovery.service.model.entities.InstanceValueMetadata;

@Getter
@Builder
@EqualsAndHashCode
@Jacksonized
public class InstanceValueMetadataDto {

    @NotBlank
    private final String displayName;

    @NotNull
    private final InstanceValueMetadata.Type type;

    @NotNull
    private final String key;

}

File: src/main/kotlin/no/novari/flyt/discovery/service/model/dtos/InstanceValueMetadataDto.kt
package no.novari.flyt.discovery.service.model.dtos

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import no.novari.flyt.discovery.service.model.entities.InstanceValueMetadata

data class InstanceValueMetadataDto(
    @field:NotBlank
    val displayName: String? = null,
    @field:NotNull
    var type: InstanceValueMetadata.Type? = null,
    @field:NotBlank
    val key: String? = null,
)
